.github/codeql/queries/assert-pure.ql

```diff
@@ -0,0 +1,29 @@
+/**
+ * @name Assert Pure
+ * @kind problem
+ * @problem.severity error
+ * @id vscode-codeql/assert-pure
+ * @description This query asserts that the modules tested in the pure-tests are "pure".
+ * This means that they don't have any transitive reference to vscode.
+ */
+import javascript
+
+class VSCodeImportDeclaration extends ImportDeclaration {
+  VSCodeImportDeclaration() {
+    this.getImportedPath().getValue() = "vscode"
+  }
+}
+
+ImportDeclaration transitiveImports(Module m) {
+  exists (ImportDeclaration i |
+    (result = i and i = m.getAnImport()) or
+    (result = transitiveImports(m.getAnImport().resolveImportedPath()))
+  )
+}
+
+from Module m, VSCodeImportDeclaration v
+where
+  // For now, let's use `src` and force this to fail. Should change to `pure`.
+  m.getFile().getParentContainer().getBaseName() = "pure" and
```

Comment:
This won't handle subfolders within `pure`. Should we consider those too?

Reply:
We don't have any at the moment, but we probably should.

Reply:
Pushed a change that should get this working on sub-directories.